### ISSUE TITLE
spark tab in rstudio preview server shows many spark tables and there is no search

### ISSUE THREAD

**Issue by rvij** [rstudio/sparklyr#266]
We installed sparklyr on rstudio server pro pointing to the cluster using yarn-client.
On inspecting the spark tab in rstudio UI we see many spark objects that are not even created by the logged in user. 

Moreover there is no search functionality to search for the object. 
Please add search and show only the spark objects created by the logged in user.

Please see the attached screen-shot.
<img width="598" alt="sparktab" src="https://cloud.githubusercontent.com/assets/2610890/19332745/ed2a461c-90a3-11e6-9e75-6971ed92c1f4.png">

**Comment by javierluraschi** [comment]
@rvij thanks for the feedback. I'm trying to understand a bit more what would be a good way to implement this. A couple questions: Are there multiple hive databases in your organization? Is the user listed under the table metadata? For the second question, please you run the following code for one of the tables to give us some sense of the type of metadata systems like yours are tracking:

```{r}
# sc <- spark_connect(master = "")
sparklyr:::df_from_sql(sc, "SHOW TBLPROPERTIES atest_10034")
```

**Comment by rvij** [comment]
Hi Javier,

 Yes that’s correct, our hadoop cluster has 100s of Hive databases. But the specific user can only query few of them.
Rest of them are not readable, so user cannot query those tables/ hive databases, we use MapR distribution of hadoop.

I get following results form the query.


> sparklyr:::df_from_sql(sc, "SHOW TBLPROPERTIES atest_10034")
# A tibble: 1 x 1
                             result
                              <chr>
1 transient_lastDdlTime\t1476278357


Since there are large number of objects showing up and no search in the spark tab, so I cannot query for my objects I loaded into spark.

Hope this helps.


Rajeev Vij

**Comment by javierluraschi** [comment]
@rvij any chance you know the SQL query you would use to filter the tables assigned to each user? I'll mark this as feature request to, at least, consider searching by table name in the UI. However, at this moment in time, it's less clear to me how `sparklyr` could filter out the tables created by the given user. Thoughts and suggestions welcome.

**Comment by rvij** [comment]
Thanks yes please include search feature to start with. In order to form a query, I need to know what does sparklyr show in the “spark tab” in Rstudio server?
Does it show all the hive objects/tables in the hive metadata? If that is the case then why do we need to show it in first place. If user would like to process hive tables using spark then he should explicitly create spark table using sparklyr.

**Comment by javierluraschi** [comment]
@rvij it currently runs this sql query: `"SHOW TABLES"` through the `HiveContext.sql()` for Spark < 2.0 and the `SessionContext.sql()` for Spark 2.0.

If we find no way to query this from Hive, we could track our own list of created tables but this would reset when the rsession restarts. Thoughts welcomed.

**Comment by Brett-A** [comment]
Hi @rvij, we are having a similar issue. We're running RStudio on an edge node and connecting to Spark via sparklyr to a MapR Hadoop cluster. However I'm relatively new to this, here is what we are running to connect to spark: 

```
library(sparklyr)
sc <- spark_connect(master = "spark://node04:7077") 
```

Node04 is a node on the Hadoop cluster that has a spark master running. We connect, see several files that were not created by the user, and get an error when clicking to preview one of the files:

```
Error in if (ncol(df) > 0) df else invisible(df) : 
  argument is of length zero
In addition: Warning message:
In value[[3L]](cond) :
  org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1.0 failed 4 times, most recent failure: Lost task 0.3 in stage 1.0 (TID 4, node02): java.io.IOException: Error getting user info for current user, brett01
	at com.mapr.fs.MapRFileSystem.lookupClient(MapRFileSystem.java:665)
	at com.mapr.fs.MapRFileSystem.lookupClient(MapRFileSystem.java:702)
	at com.mapr.fs.MapRFileSystem.getMapRFileStatus(MapRFileSystem.java:1432)
	at com.mapr.fs.MapRFileSystem.getFileStatus(MapRFileSystem.java:1056)
	at org.apache.hadoop.hive.ql.io.RCFile$Reader.<init>(RCFile.java:1343)
	at org.apache.hadoop.hive.ql.io.RCFileRecordReader.<init>(RCFileRecordReader.java:100)
	at org.apache.hadoop.hive.ql.io.RCFileInputFormat.getRecordReader(RCFileInputFormat.java:57)
	at org.apache.spark.rdd.HadoopRDD$$anon$1.<init>(HadoopRDD.scala:237)
	at org.apache.spark.rdd.HadoopRDD.compute(HadoopRDD.scala:208)
	at org.apache.spark.rdd.HadoopRDD.compute(HadoopRDD.scala:101)
	at org.apache.spark.r [... truncated]
```

Also, does the Spark master need to be localized to the client/edge node, or R-Server localized to the node with the spark master? Any insights to how you configured Sparklyr with RStudio and a MapR distro of Spark will be helpful too, thanks

**Comment by javierluraschi** [comment]
@rvij search functionality is now implemented in rstudio daily builds (until rstudio 1.1 is released), requires using `sparklyr` from github (until `sparklyr 0.6` gets released to CRAN).

**Comment by javierluraschi** [comment]
@rvij see https://github.com/rstudio/rstudio/issues/1549, will close this one since this is now tracked under the RStudio repo.
